### DISCUSSION TITLE
MoveFile

### DISCUSSION THREAD

**Discussion by Liuyadi-Code** [dokan-dev/dokany#1212] [Q&A] [answered]
Ask which file system API is triggered when files in dokany are moved to another disk

**Comment by Olof-Lagerkvist** [comment, marked as answer]
There is no specific function for that. Files moved to other drives are internally read with ReadFile calls and written with WriteFile calls.

**Reply by Liuyadi-Code** [reply]
Now I want to make an https request to write the file when moving the file to a disk. Can I do that? How do I capture this drag operation?

**Reply by Olof-Lagerkvist** [reply]
You might be able to detect it by checking for read calls followed by a delete call or something like that. But it will not be very reliable. I would rather suggest a shell extension for this. It will only work if the file is moved in or between Windows Explorer windows though and not for example from command line. But it is a lot easier to detect file moves specifically. You could probably start with some existing shell extension project and adjust it to do what you want. For example: https://github.com/broken-e/DragDropConfirm

**Reply by Liuyadi-Code** [reply]
Thank you very much for your answer, I wish you a happy life!
